eng/references.targets

```diff
@@ -36,32 +54,41 @@
     </ItemGroup>
   </Target>
 
-  <Target Name="AddDefaultTestReferences" BeforeTargets="SetupDefaultReferences" Condition="'$(IsTestProject)' == 'true' or '$(IsTestSupportProject)' == 'true'">
+  <!-- Filter out conflicting targeting pack references -->
+  <Target Name="RemoveTargetingPackReferenceExclusions"
+          Condition="'$(DisableImplicitFrameworkReferences)' != 'true'"
+          AfterTargets="ResolveTargetingPackAssets">
+    <!-- Filter out the current assembly and its P2Ps. -->
     <ItemGroup>
-      <DefaultReferenceExclusions Include="@(ReferenceFromRuntime)"/>
-
-      <!-- Reference everything in the targeting pack directory -->
-      <DefaultReferenceDirs Include="$(RefPath)" />
-      <DefaultReferenceItems Include="%(DefaultReferenceDirs.Identity)/*.dll" />
+      <TargetingPackReferenceExclusion Include="$(TargetName)" />
+      <TargetingPackReferenceExclusion Include="@(ProjectReference->'%(Filename)')" />
+    </ItemGroup>
 
-      <DefaultReferenceExclusions>
-        <RefDir>%(DefaultReferenceDirs.Identity)</RefDir>
-      </DefaultReferenceExclusions>
-      <_defaultReferenceExclusionsFullPath Include="%(DefaultReferenceExclusions.RefDir)%(DefaultReferenceExclusions.Identity).dll" />
+    <!-- Filter out contract assemblies that we still live build: https://github.com/dotnet/runtime/issues/34253. -->
+    <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETStandard' and $(TargetFrameworkVersion.StartsWith('v2.'))">
+      <TargetingPackReferenceExclusion Include="System.ComponentModel.Composition" />
+      <TargetingPackReferenceExclusion Include="System.Reflection.DispatchProxy" />
+    </ItemGroup>
 
-      <!-- Ensure conflict resolution can see these references. -->
-      <Reference Include="%(DefaultReferenceItems.FullPath)" Private="false" />
+    <ItemGroup>
+      <_targetingPackReferenceWithExclusion Include="@(Reference)">
+        <Exclusion>%(TargetingPackReferenceExclusion.Identity)</Exclusion>
+      </_targetingPackReferenceWithExclusion>
+      <Reference Remove="@(_targetingPackReferenceWithExclusion)"
+                 Condition="'%(_targetingPackReferenceWithExclusion.ExternallyResolved)' == 'true' and '%(_targetingPackReferenceWithExclusion.Filename)' == '%(_targetingPackReferenceWithExclusion.Exclusion)'" />
     </ItemGroup>
   </Target>
 
-  <Target Name="RemoveConflictResolutionAssetsForTests"
+  <Target Name="RemoveDefaultReferenceExclusions"
```

Comment:
when will we be using this target ?

Reply:
That target is already in master: https://github.com/dotnet/runtime/blob/master/eng/references.targets#L57, I just renamed it and modified it slightly.

Its purpose is to remove references that are added from the RefPath.